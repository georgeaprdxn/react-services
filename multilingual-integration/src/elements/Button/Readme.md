Button

```jsx inside Markdown
const clickHandler = () => alert('Clicked')
;<Button title="Sample Title" onClick={clickHandler} />
```
