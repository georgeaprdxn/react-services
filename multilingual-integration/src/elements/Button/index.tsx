import React, { FunctionComponent } from 'react'

import { StyledButton } from './styled'

type ButtonProps = {
	/**
	 * class name for the button element
	 */
	buttonClass?: string
	/**
	 * handles button clicks
	 */
	onClick?:
		| ((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void)
		| undefined
	/**
	 * button's title attribute
	 */
	title: string
	/**
	 * flag to disable the button
	 */
	isDisabled?: boolean
}

const Button: FunctionComponent<ButtonProps> = ({
	buttonClass,
	onClick,
	title,
	isDisabled,
}) => {
	return (
		<StyledButton
			className={buttonClass}
			onClick={onClick}
			title={title}
			disabled={isDisabled}
		>
			{title}
		</StyledButton>
	)
}
Button.defaultProps = {
	isDisabled: false,
	buttonClass: 'btn-primary',
}

export default Button
