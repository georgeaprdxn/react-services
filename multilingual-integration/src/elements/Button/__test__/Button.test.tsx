import React from 'react'
import { shallow } from 'enzyme'

import Button from '../index'

describe('Button element tests', () => {
	let wrapper: null | any = null
	beforeEach(() => {
		wrapper = shallow(<Button title="Some title" />)
	})
	it('should render a <button />', () => {
		expect(wrapper.find('.btn-primary').length).toEqual(1)
	})

	it('matches the snaphot', () => {
		expect(wrapper).toMatchSnapshot()
	})
})
