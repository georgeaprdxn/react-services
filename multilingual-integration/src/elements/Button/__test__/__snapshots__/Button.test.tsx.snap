// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button element tests matches the snaphot 1`] = `
<styled.button
  className="btn-primary"
  disabled={false}
  title="Some title"
>
  Some title
</styled.button>
`;
